client/java/src/main/java/com/linecorp/centraldogma/client/AbstractCentralDogmaBuilder.java

```diff
@@ -128,18 +141,38 @@ protected final boolean isUseTls() {
         return useTls;
     }
 
+    /**
+     * Sets the paths to look for to read the {@code .json} file that contains the client profiles.
+     * The paths are tried in the order of iteration. The default value of this property is
+     * <code>[ {@value #TEST_PROFILE_RESOURCE_PATH}, {@value #PROFILE_RESOURCE_PATH} ]</code>, which means
+     * the builder will check if {@value #TEST_PROFILE_RESOURCE_PATH} exists first and will try
+     * {@value #PROFILE_RESOURCE_PATH} only if {@value #TEST_PROFILE_RESOURCE_PATH} is missing.
+     */
+    public final B profileResources(String... paths) {
+        return profileResources(ImmutableList.copyOf(requireNonNull(paths, "paths")));
+    }
+
+    /**
+     * Sets the paths to look for to read the {@code .json} file that contains the client profiles.
+     * The paths are tried in the order of iteration. The default value of this property is
+     * <code>[ {@value #TEST_PROFILE_RESOURCE_PATH}, {@value #PROFILE_RESOURCE_PATH} ]</code>, which means
+     * the builder will check if {@value #TEST_PROFILE_RESOURCE_PATH} exists first and will try
+     * {@value #PROFILE_RESOURCE_PATH} only if {@value #TEST_PROFILE_RESOURCE_PATH} is missing.
+     */
+    public final B profileResources(Iterable<String> paths) {
+        final List<String> newPaths = ImmutableList.copyOf(requireNonNull(paths, "paths"));
```

Comment:
nit: How about checking `selectedProfile == null` and `hosts.isEmpty()`? (or resetting them because profile resources are to be changed here?)

Reply:
Good point. Added a check for `selectedProfile == null`. I didn't check if `hosts.isEmpty()` though because a user is allowed add an individual host using `host()` at any time.